py-wacz/wacz/waczindexer.py

```diff
@@ -149,12 +154,28 @@ def extract_page_lists(self, lists):
 
             self.extra_page_lists[uid] = text_list
 
+    def match_detected_pages(self, detected_pages, passed_pages_url, passed_pages_ts):
+        for page in detected_pages:
+            page = detected_pages[page]
+            url = page["url"]
+            ts = page["timestamp"]
+            if passed_pages_url == url and passed_pages_ts == None:
+                return page
+            if passed_pages_url == url and passed_pages_ts == ts:
+                return page
+        return 0
+
     def check_pages_and_text(self, record):
         url = record.rec_headers.get("WARC-Target-URI")
         date = record.rec_headers.get("WARC-Date")
         ts = iso_date_to_timestamp(date)
         id_ = ts + "/" + url
-
+        if id_ in self.passed_pages_dict.keys():
+            del self.passed_pages_dict[id_]
+            self.pages[id_] = {"timestamp": ts, "url": url, "title": url}
+        if url in self.passed_pages_dict.keys():
+            del self.passed_pages_dict[url]
+            self.pages[id_] = {"timestamp": ts, "url": url, "title": url}
```

Comment:
We should see if the `passed_pages_dict` entry has a `title` or `text` and add those, if they're set